### PR TITLE
Support path prefixes

### LINKED ISSUE #2: Double slash if the base URI has a path
To reproduce:

```scala
//> using dep "tech.neander::smithy4s-deriving::0.0.2"
//> using dep tech.neander::smithy4s-fetch::0.0.3
//> using platform scala-js
//> using scala 3.4.2
//> using option -Wunused:all

import smithy4s.*

import scala.annotation.experimental
import scala.scalajs.js.Promise

import deriving.given
import deriving.*
import aliases.*
import smithy4s_fetch.SimpleRestJsonFetchClient

@experimental
trait HttpbinService derives API:

  @readonly
  @httpGet("/get")
  def get(): Promise[Document]

@main @experimental
def helloWorld =
  val service: HttpbinService =
    SimpleRestJsonFetchClient(
      API.service[HttpbinService],
      "https://httpbin.org/api",
    ).make.unliftService

  service.get().`then`(v => println(v))
```

```html
<!DOCTYPE html>

<html>
  <head>
    <script src="./helloWorld.js" type="text/javascript"></script>
  </head>
</html>
```

Open the HTML file, see network tab in the browser, look for the request path.

- **Expectation**: `https://httpbin.org/api/get`
- **Reality**: `https://httpbin.org//api/get`

### PR DESCRIPTION
**Pull request by kubukoz**
Closes #2

### CODE REVIEW

File: smithy4s-fetch.scala [hunk 1]
```diff
@@ -233,7 +233,14 @@ private[smithy4s_fetch] case class SimpleRestJsonCodecs(
       uriScheme,
       uri.host,
       uri.port.toIntOption,
-      uri.pathname.split("/"),
+      uri.pathname
+        // drop the guaranteed leading slash, so that we don't produce an empty segment for it
```
Comment (keynmol) [inline smithy4s-fetch.scala:237]: Hm, I think the "guarantee" is not strong enough for me - it's not a `NonEmptyList` after all.

I think it's better to do `url.pathname.split("/").dropWhile(_.isEmpty)` - it should be safer.

Comment (kubukoz) [reply smithy4s-fetch.scala:237]: I took the guarantee from the Scaladoc, https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname also repeats this claim... but yeah I can still add that

Comment (kubukoz) [reply smithy4s-fetch.scala:237]: actually, dropWhile will drop empty segments if they precede non-empty ones, I think that'll break a test or two. Having a guaranteed slash before each segment is one way to ensure we don't accidentally drop segments. The spec seems pretty unambiguous on that part:

> For each segment of url’s path: append U+002F (/) followed by segment to output

Comment (keynmol) [reply smithy4s-fetch.scala:237]: right, I sort of hoped empty segments were disallowed in general 🙃

**Review by keynmol — APPROVED** [review]
Thanks a lot, especially for the tests!

**Pull request merged**